### DISCUSSION TITLE
overall_average_cost not a number?

### DISCUSSION THREAD

**Discussion by ccristal** [BottlecapDave/HomeAssistant-OctopusEnergy#419] [Q&A] [answered]
Hi!
I have this condition in my automation:

```
condition: numeric_state
entity_id: binary_sensor.octopus_energy_target_cheapest_intermittent_overnight_hour
attribute: overall_average_cost
below: sensor.octopus_energy_electricity_xxxxxxxxxx_yyyyyyyyyyyyy_export_current_rate
```

When this condition evaluates, I get the following error:

`Error: In 'numeric_state': In 'numeric_state' condition: entity binary_sensor.octopus_energy_target_cheapest_intermittent_overnight_hour state '7.17675p' cannot be processed as a number`

![image](https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/assets/5416417/9517bccd-9608-4d4c-976a-971756a928cf)

What am I doing wrong?

Thank you!

**Comment by BottlecapDave** [comment, marked as answer]
Hello there. Unfortunately, the attributes are currently outputted as strings as they required the unit value (i.e. `p`) with them to reduce confusion. I'll look into if there is a better way of doing this, or providing a `raw` representation like I do elsewhere. For now, you can use the following template to remove the unit value and convert the string into a float

```
{{ state_attr('binary_sensor.octopus_energy_target_cheapest_intermittent_overnight_hour', 'overall_average_cost')[:-1] | float }}
```
